src/zlib.zig

```diff
@@ -429,7 +429,7 @@ pub const ZlibReaderArrayList = struct {
 
     pub fn alloc(ctx: *anyopaque, items: uInt, len: uInt) callconv(.C) *anyopaque {
         var this = @as(*ZlibReader, @ptrCast(@alignCast(ctx)));
-        const buf = this.allocator.alloc(u8, items * len) catch unreachable;
+        const buf = this.arena.allocator().alloc(u8, items * len) catch unreachable;
```

Comment:
We shouldn’t need to use the arena here. Zlib shouldn’t leak memory. It likely means we are missing a function call. 

arenas are really bad for performance when used inside arrays because zig’s arena allocator only frees at the end. So we are keeping a duplicate copy of the entire array on every allocation

Reply:
changed to use mimalloc directly, but reporting back to HTTP client is still pending.